### PR TITLE
Implement connection timeout for Algolia connection

### PR DESCRIPTION
**Pull request by jupeter**
Summary: 
 - doesn't BC break
 - Rebased
 - Doesn't have unit test (currently your tests aren't organized).

Thank you for merge and not wait 7 months.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,9 +101,11 @@ There's two optional parameters you can add to this file:
 algolia:
     catch_log_exceptions: true
     index_name_prefix: staging
+    connection_timeout: 2
 ```
 * **catch_log_exceptions**: If set to true, all exceptions thrown in the doctrine event subscriber will be caught and logged.
 * **index_name_prefix**: If set, this will add a prefix to all the index names (Useful if you want to setup multiple environments within the same Algolia app)
+* **connection_timeout**: If set, this will set connection timeout to Algolia
```
Comment (Haroenv) [inline README.md]: add "in seconds"

File: Indexer/Indexer.php [hunk 2]
```diff
@@ -104,11 +104,12 @@ public function setIndexNamePrefix($indexNamePrefix)
         $this->indexNamePrefix = $indexNamePrefix;
     }
 
-    public function setApiSettings($application_id, $api_key)
+    public function setApiSettings($application_id, $api_key, $connection_timeout)
```
Comment (rayrutjes) [inline Indexer/Indexer.php]: $connection_timeout should have a default value given it didn't exist before and it is part of the public interface.

**Comment by rayrutjes** [comment]
@julienbourdeau would you be able to review & merge it today or tomorrow?

File: Indexer/Indexer.php (hunk 2 above)
Comment (jupeter) [inline Indexer/Indexer.php]: I set parameter as optional.

File: Indexer/Indexer.php [hunk 3]
```diff
@@ -640,8 +644,13 @@ public function getClient()
                 $this->apiSettings['application_id'],
                 $this->apiSettings['api_key']
             );
+
+            $this->client->setConnectTimeout(
+                $this->apiSettings['connection_timeout']
```
Comment (julienbourdeau) [inline Indexer/Indexer.php]: You will get an _undefined index_ notice here if the `setApiSettings` method was called without the new `$connection_timeout` parameter.

I guess this method should be called only if `$this->apiSettings['connection_timeout']` is not set.

File: README.md [hunk 4]
```diff
@@ -101,9 +101,11 @@ There's two optional parameters you can add to this file:
 algolia:
     catch_log_exceptions: true
     index_name_prefix: staging
+    connection_timeout: 2
```
Comment (julienbourdeau) [inline README.md]: I think the default value is 1 second in the algoliasearch-client-php library. We should probably leave at one, or even default it to null.

Comment (julienbourdeau) [inline README.md]: Sorry I realized it was in the README.md. For some reasons I was convinced it was in the `service.yml`. My bad!

**Comment by julienbourdeau** [comment]
@jupeter I made a small modification according to my previous comments. Please have a look at this  PR #106. If it's fine with you, we'll merge it but if you prefer, you can update this pull request.

**Pull request closed without merging**